yt_dlp/_globals.py

```diff
@@ -0,0 +1,28 @@
+from collections import defaultdict
+
+# Internal only - no backwards compatibility guaranteed
+
+
+class Indirect:
+    def __init__(self, initial, /):
+        self.value = initial
+
+    def __repr__(self, /):
+        return f'{type(self).__name__}({self.value!r})'
```

Comment:
We'll want to have other `ContextVars` in this module. If `Indirect` had a similar `.get()`/`.set()` interface, devs wouldn't have to keep track which is which. Ref: https://github.com/yt-dlp/yt-dlp/pull/7426/files#diff-9219d6ec74a05fa4f50ab68885c42dc10bbfcbed4219090e7e40f38cc7bbd2f3R8

Reply:
Why would these have to be contextvars? We would not want to have different values for each thread, would we?

Reply:
To clarify: The vars in this PR should **not** be `ContextVars`.

But future additions like `ffmpeg_location`, `verbosity` etc would need to be.

Reply:
No they couldn't? If you want to keep compatibility, verbosity could not be a global anyways...

The value is suppose to be unique to each YoutubeDL instance, and therefore should carry over thread barriers, but NOT over YoutubeDL instances. It would only be useful if one instance were running at a time, otherwise we would break output compatibility for multiple instances.

I do have a somewhat different implementation in mind that will expose the same `.value`. Every instance of `YoutubeDL` can register itself as its `id` at `verbose`, and if the value is requested it can walk up the stack and see if a function from a registered `YoutubeDL` was called to give the correct value based on that. That will preserve full compatibility, but will likely only be implemented further into the future.

Even if that were not the case though, we can do wrap a contextvar:
```py
class IndirectContextVar:
    def __init__(self, name: str, /, *, default=None):
        self._value = ContextVar(name, default)

    @property
    def value(self, /):
        self._value.get()

    @value.setter
    def value(self, value, /):
        self._value.set(value)
```

Reply:
> Even if that were not the case though, we can do wrap a contextvar:

Fair enough